tests/testOpsimFieldSlicer.py

```diff
@@ -116,10 +118,9 @@ def tearDown(self):
     def testWarning(self):
         self.testslicer.setupSlicer(self.simData, self.fieldData)
         with warnings.catch_warnings(record=True) as w:
-            warnings.simplefilter('always')
             self.testslicer.setupSlicer(self.simData, self.fieldData)
-            assert len(w) == 1
-            assert "Re-setting up an OpsimFieldSlicer" in str(w[-1].message)
+        self.assertEqual(len(w), 1)
+        self.assertIn("Re-setting up an OpsimFieldSlicer", str(w[-1].message))
```

Comment:
Learn something every day. I had assumed that when the `with` block ended that the context manager exit handler would run which could in theory tidy up the object associated with the `w`.